Explain how to change API identifier for doctrine entities (#1086)
api-platform/docs: core/identifiers.md

Your `PersonDataProvider` will now work as expected!

## Changing Identifier in a Doctrine Entity

If your resource is also a Doctrine entity and you want to use another identifier other than the Doctrine one, you have to unmark it:

```php
<?php
namespace App\Entity;

> use App\Uuid;

Author: ```suggestion
use ApiPlatform\Core\Annotation\ApiResource;
use App\Uuid;
```

Your `PersonDataProvider` will now work as expected!

> ### Changing identifier in a Doctrine entity

Reviewer: ```suggestion
## Changing Identifier in a Doctrine Entity
```

> If your ApiResource is also a Doctrine entity, you have to unmark its id as identifier:

Reviewer: ```suggestion
If your resource is also a Doctrine entity and you want to use another identifier as the Doctrine one, you have to unmark it:
```

Author: @alanpoulain I don't think this is the case, is it? I don't want to use it as the Doctrine one, but as the ApiPlatform one.

Reviewer: Yes, I made a mistake, updated.

```php
<?php
namespace App\Entity;

> use App\Uuid;

Reviewer: Missing `ApiResource`.

Author: it is not in the existing example either, should I update both?

Reviewer: Yes you can :slightly_smiling_face:

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 * @ApiResource
 */
final class Person
{

> 

Reviewer: Line to remove.

Author: ```suggestion

```

    /**
     * @var int
     * @ApiProperty(identifier=false)
     *
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;
    
    /**
     * @var Uuid
     * @ApiProperty(identifier=true)
     */
    public $code;
    
    // ...
}
```

> 

Reviewer: Line to remove.

Your `PersonDataProvider` will now work as expected!

## Changing Identifier in a Doctrine Entity

If your resource is also a Doctrine entity and you want to use another identifier other than the Doctrine one, you have to unmark it:

```php
<?php
namespace App\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use App\Uuid;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 * @ApiResource
 */
final class Person
{
    /**
     * @var int
     * @ApiProperty(identifier=false)
     *
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;
    
    /**
     * @var Uuid

>      * @ApiProperty(identifier=true)

Reviewer 2: It should have a column definition too.

Author: @GregoireHebert thanks, added
